### PR TITLE
ARROW-1580: [Python] Instructions for setting up nightly builds on Linux

### PR DESCRIPTION
**Pull request by cpcloud**

### CODE REVIEW

File: cpp/README.md [hunk 1]
```diff
@@ -279,7 +279,83 @@ both of these options would be used rarely. Current known uses-cases when they a
 
 *  Parameterized tests in google test.
 
+## Nightly Builds of `arrow-cpp`, `parquet-cpp`, and `pyarrow` for Linux
```
Comment (wesm) [inline cpp/README.md]: Need to think about the best place to put this documentation. Maybe in the Python Sphinx area

File: cpp/README.md [hunk 2]
```diff
@@ -279,7 +279,83 @@ both of these options would be used rarely. Current known uses-cases when they a
...
+
+Nightly builds of conda packages are available for [`arrow-cpp`][5],
+[`parquet-cpp`][6], and [`pyarrow`][7].
```
Comment (wesm) [inline cpp/README.md]: We aren't allowed to advertise these in the project per ASF policies https://issues.apache.org/jira/browse/ARROW-1935

**Review by wesm — COMMENTED** [review]
Thanks for doing this! couple comments -- I will also try to make a run-through of this from scratch

File: cpp/README.md (hunk 2 above)
Comment (cpcloud) [reply cpp/README.md]: Cool, will remove.

File: cpp/README.md (hunk 1 above)
Comment (cpcloud) [reply cpp/README.md]: Yeah, I kept it here because it's related to all major arrow components not just Python

Comment (wesm) [reply cpp/README.md]: We're running into the issue that we need to evolve the main Arrow site into cross-project documentation. Maybe we can do that when the site becomes a little better organized

Comment (xhochy) [reply cpp/README.md]: We could try to use https://breathe.readthedocs.io/en/latest/ to merge the C++ and Python documentation. As they are both very tightly coupled, I think they should have a common documentation. For example, the installation instructions are nearly the same.

Comment (wesm) [reply cpp/README.md]: I'd be up for that. We could create a top-level doc directory. We also need to start writing Java documentation that is not Javadoc

**Review by xhochy — APPROVED** [review]

**Comment by wesm** [comment]
+1, thanks @cpcloud!

**Pull request closed without merging**